app/models/filter.rb

```diff
@@ -158,19 +162,24 @@ def enforce_inherited_taxonomies
   end
 
   def inherit_taxonomies!
-    self.organization_ids = role.organization_ids if allows_organization_filtering?
-    self.location_ids = role.location_ids if allows_location_filtering?
+    self.organization_ids = role.organization_ids if resource_taxable_by_organization?
+    self.location_ids = role.location_ids if resource_taxable_by_location?
     build_taxonomy_search
   end
 
   private
 
   def build_taxonomy_search
-    orgs = build_taxonomy_search_string('organization')
-    locs = build_taxonomy_search_string('location')
-
-    orgs = [] if !granular? || !resource_class.allows_organization_filtering?
-    locs = [] if !granular? || !resource_class.allows_location_filtering?
+    orgs = if resource_taxable_by_organization?
+             build_taxonomy_search_string('organization')
+           else
+             ''
+           end
+    locs = if resource_taxable_by_location?
+             build_taxonomy_search_string('location')
+           else
+             ''
+           end
```

Comment:
Now that I see it here, should `build_taxonomy_search_string` instead return `''` if it's not taxable?

Reply:
looks much cleaner now :)